src/explore-education-statistics-common/src/components/form/__tests__/FormFieldTextArea.test.tsx

```diff
@@ -0,0 +1,318 @@
+    );
+
+    expect(screen.getByLabelText('Test field')).toHaveAttribute(
+      'aria-describedby',
+      'testField-hint',
+    );
+    expect(screen.getByText('Test hint')).toHaveAttribute(
+      'id',
+      'testField-hint',
+    );
+  });
+
+  test('renders with correct defaults ids with form', () => {
+    render(
+      <Formik initialValues={{}} onSubmit={noop}>
+        <Form id="testForm">
+          <FormFieldTextArea
+            name="testField"
+            label="Test field"
+            hint="Test hint"
+          />
+        </Form>
+      </Formik>,
+    );
+
+    expect(screen.getByLabelText('Test field')).toHaveAttribute(
+      'id',
+      'testForm-testField',
+    );
+
+    expect(screen.getByLabelText('Test field')).toHaveAttribute(
+      'aria-describedby',
+      'testForm-testField-hint',
+    );
+
+    expect(screen.getByText('Test hint')).toHaveAttribute(
+      'id',
+      'testForm-testField-hint',
+    );
+  });
+
+  test('renders with correct defaults ids without form', () => {
+    render(
+      <Formik initialValues={{}} onSubmit={noop}>
+        <FormFieldTextArea
+          name="testField"
+          label="Test field"
+          hint="Test hint"
+        />
+      </Formik>,
+    );
+
+    expect(screen.getByLabelText('Test field')).toHaveAttribute(
+      'id',
+      'testField',
+    );
+
+    expect(screen.getByLabelText('Test field')).toHaveAttribute(
+      'aria-describedby',
+      'testField-hint',
+    );
+
+    expect(screen.getByText('Test hint')).toHaveAttribute(
+      'id',
+      'testField-hint',
+    );
+  });
+
+  test('renders with correct custom ids with form', () => {
+    render(
+      <Formik initialValues={{}} onSubmit={noop}>
+        <Form id="testForm">
+          <FormFieldTextArea
+            id="customId"
+            name="testField"
+            label="Test field"
+            hint="Test hint"
+          />
+        </Form>
+      </Formik>,
+    );
+
+    expect(screen.getByLabelText('Test field')).toHaveAttribute(
+      'id',
+      'testForm-customId',
+    );
+
+    expect(screen.getByLabelText('Test field')).toHaveAttribute(
+      'aria-describedby',
+      'testForm-customId-hint',
+    );
+
+    expect(screen.getByText('Test hint')).toHaveAttribute(
+      'id',
+      'testForm-customId-hint',
+    );
+  });
+
+  test('renders with correct custom ids without form', () => {
+    render(
+      <Formik initialValues={{}} onSubmit={noop}>
+        <FormFieldTextArea
+          id="customId"
+          name="testField"
+          label="Test field"
+          hint="Test hint"
+        />
+      </Formik>,
+    );
+
+    expect(screen.getByLabelText('Test field')).toHaveAttribute(
+      'id',
+      'customId',
+    );
+
+    expect(screen.getByLabelText('Test field')).toHaveAttribute(
+      'aria-describedby',
+      'customId-hint',
+    );
+
+    expect(screen.getByText('Test hint')).toHaveAttribute(
+      'id',
+      'customId-hint',
+    );
+  });
```

Comment:
Bit torn about a lot of these tests as they're really repetitive across a lot of the field components. Appreciate I probably wrote the first components that did this type of checking, but regretting it a little now!

In these tests, we're just checking that a correct `id` has been passed to the component rendered by `FormField`. We (usually) check hint and error ids align correctly to the `id` prop in the `Form*` (non-field) component tests too, so this extra checking is basically redundant.

We don't have to do it in this PR, but perhaps we could do a follow up to refactor the Formik and RHF component tests and move any relevant tests into `FormField` and `RHFFormField` test suites respectively.

Tests would be something like the following:

```tsx
describe('FormField', () => {
  const TestInput = ({
    error,
    id,
    label,
    name,
    ...props
  }: { error?: string; label: string } & HTMLAttributes<HTMLInputElement>) => {
    return (
      <>
        <label htmlFor={id}>{label}</label>

        {error && <p>{error}</p>}

        <input
          // eslint-disable-next-line react/jsx-props-no-spreading
          {...props}
          name={name}
          id={id}
        />
      </>
    );
  };

  test('renders field with default id', () => {
    render(
      <Formik initialValues={{}} onSubmit={noop}>
        <FormField name="testField" label="Test field" as={TestInput} />
      </Formik>,
    );

    expect(screen.getByLabelText('Test field')).toHaveAttribute(
      'id',
      'testField',
    );
  });

  test('renders field with default id with form', () => ...);

  test('renders field with default id without form', () => ...);

  test('renders field with custom id with form', () => ...); 

  test('renders field with custom id without form', () => ...);

  test('renders error message', () => ...);

  // etc, etc...
});
```

Note that this means we don't really need to check stuff like hints anymore as we can leave this to the base `Form*` component tests instead. 

Would probably suggest keeping tests more relevant to proving integration or behaviours like changing / blurring the field in the specific `FormField*` and `RHFFormField*` components

Any thoughts? We can also just keep doing what we're doing. That's fine too if we don't mind the repetitiveness of it.

Reply:
TBH I only added them because I thought you'd say they should be there to be consistent with the other FormField* tests 😆 
I'll add a note to the RHF form ticket to look at improving the tests as part of that.